File: client/src/types/logs.ts
/**
 * TypeScript types for Logs API
 */

export type SeverityLevel = 'DEBUG' | 'INFO' | 'WARN' | 'ERROR' | 'FATAL';

export type TimeRange = '5m' | '1h' | '1d' | '1w';

export type SearchMode = 'simple' | 'advanced';

export interface LogEntry {
  event_name: string;
  trace_id: string;
  span_id: string;
  log_timestamp: string;        // ISO 8601 timestamp
  observed_timestamp: string;   // ISO 8601 timestamp
  severity_text: SeverityLevel | string;  // Can be any string if not one of the standard levels
  body: string;
  service_name: string;
  attributes: Record<string, any>;  // Parsed from JSON string
}

export interface LogsResponse {
  logs: LogEntry[];
  total_count: number;
  page: number;
  page_size: number;
  has_more: boolean;
  severity_counts: Record<string, number>;
}

export interface SeverityTimelinePoint {
  timestamp: string;  // ISO 8601 timestamp
  ERROR: number;
  WARN: number;
  INFO: number;
  DEBUG: number;
}

export interface SeverityTimelineResponse {
  timeline: SeverityTimelinePoint[];
  service_name: string;
  time_range: string;
}

export interface LogsFilters {
  service_name: string;
  time_range: TimeRange;
  search?: string;
  search_mode: SearchMode;
  severity_filter?: string;  // Comma-separated severity levels
  trace_id?: string;
  page: number;
  page_size: number;
}

File: server/models/logs.py
"""Data models for logs API."""

from pydantic import BaseModel
from typing import Optional, Dict, Any, List
from datetime import datetime


class LogEntry(BaseModel):
    """Individual log entry from logs_synced table."""
    event_name: Optional[str] = ""
    trace_id: Optional[str] = ""
    span_id: Optional[str] = ""
    log_timestamp: datetime
    observed_timestamp: datetime
    severity_text: Optional[str] = "INFO"
    body: str
    service_name: str
    attributes: Dict[str, Any]  # Parsed from JSON string


class LogsResponse(BaseModel):
    """Response for logs list endpoint."""
    logs: List[LogEntry]
    total_count: int
    page: int
    page_size: int
    has_more: bool
    severity_counts: Dict[str, int]


class SeverityTimelinePoint(BaseModel):
    """Single point in severity timeline."""
    timestamp: datetime
    ERROR: int
    WARN: int
    INFO: int
    DEBUG: int


class SeverityTimelineResponse(BaseModel):
    """Response for severity timeline endpoint."""
    timeline: List[SeverityTimelinePoint]
    service_name: str
    time_range: str
